projects/showcase/cypress/support/e2e-driver.ts

```diff
@@ -1,198 +1,179 @@
+    snapshot: Cypress.Chainable<{
+      buffer: Buffer;
+      height: number;
+      width: number;
+    }>
+  ) => this.compareSnapshots(snapshot).shouldBeGreaterThen(0);
+}
+export class E2eDriver {
+  private helper = new CypressHelper();
 
-  // Used to reset the console.warn spy for cases when it is known that warnings have been
-  // logged so that the test will not fail (the spy is first defined in commands.ts and
-  // warnings are evaluated as failures in afterEach())
-  resetConsoleWarnings = (): E2eDriver => {
-    cy.get('@consoleWarnSpy').then((spy: any) => {
-      if (spy.callCount > 0) {
-        cy.log(`Clearing ${spy.callCount} console warning(s)...`);
-        spy.resetHistory();
-      }
-    });
-    return this;
+  beforeAndAfter = () => {
+    beforeEach(() => {});
+    afterEach(() => {});
   };
 
-  when = {
-    wait: (ms: number): E2eDriver => {
-      cy.wait(ms);
-      return this;
-    },
-    waitForFetch: (url: string, method: string = 'GET'): E2eDriver => {
-      cy.intercept(method, url).as('awaitFetch');
-      cy.wait('@awaitFetch', { timeout: 30000 });
-      return this;
-    },
-    waitForMapToIdle: (timeout: number = 30000): E2eDriver => {
-      cy.get('[data-idle="true"]', { timeout: timeout }).should('exist');
-      return this;
-    },
-    waitForMapLoaded: (timeout: number = 30000): E2eDriver => {
-      cy.get('[data-loaded="true"]', { timeout: timeout }).should('exist');
-      return this;
-    },
-    clickLanguageButton: (language: string): E2eDriver => {
-      cy.get('.lang-button').contains(language).click();
-      return this;
-    },
-    clickPopupCloseButton: (): E2eDriver => {
-      cy.get('.maplibregl-popup-close-button').click();
-      return this;
-    },
-    clickFromCodeRadioButton: (): E2eDriver => {
-      cy.get('mat-radio-button').contains('from code').click();
-      return this;
-    },
-    clickStreetsRadioButton: (): E2eDriver => {
-      cy.get('mat-radio-button').contains('streets').click();
-      return this;
-    },
-    clickCountryNamesButton: (): E2eDriver => {
-      cy.get('mat-button-toggle').contains('countries names').click();
-      return this;
-    },
-    clickZoomToBoundsButton: (): E2eDriver => {
-      cy.get('.zoom-button').click();
-      return this;
-    },
-    clickHelloCustomButton: (): E2eDriver => {
-      cy.on('window:alert', this.alertStub);
-      cy.get('.custom-control').click();
-      return this;
-    },
-    clickEnableTerrainControlButton: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain').click();
-      return this;
-    },
-    clickDisableTerrainControlButton: (): E2eDriver => {
-      cy.get('.maplibregl-ctrl-terrain-enabled').click();
-      return this;
-    },
-    clickHideControlsButton: (): E2eDriver => {
-      cy.get('button').contains('Hide Controls').click();
-      return this;
-    },
+  given = {
+    alertStub: () => cy.on("window:alert", this.helper.given.stub("alert")),
```

Comment:
Should this `cy.on` be part of an extended helper?

Reply:
Hmm..
Yes, we could add an extend helper to maplibre as well.